### PR TITLE
[1/3] Adds Databricks to UI

### PR DESCRIPTION
**Pull request by hsubbaraj-spiral**
## Describe your changes and why you are making these changes
Adds Databricks Integration to UI
## Related issue number (if any)
eng-2087
## Loom demo (if any)
Screen Shot 2022-12-16 at 10 30 15 AM

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by agiron123** [comment]
Do we nee to add another logo here besides the spark logo?

**Review by agiron123 — APPROVED** [review]
Nice work here, just left a comment about the logo that we should be using, but otherwise code here looks good with me.

**Comment by agiron123** [comment]
Looks like you are failing lint in the UI build github action. 

Please go to aqueduct/ui/common and run:
npm run lint:fix to take care of these.

**Comment by agiron123** [comment]
@hsubbaraj-spiral is this something that we want to get merged soon, or are we waiting for the rest of the spark implementation to be ready?

Also, do we need to add another icon here? Happy to provide design help and get you the icon you may need.

**Comment by nfx** [comment]
please get databricks icon from https://brand.databricks.com/

File: src/golang/go.mod [hunk 1]
```diff
@@ -3,9 +3,10 @@ module github.com/aqueducthq/aqueduct
 go 1.16
 
 require (
-	cloud.google.com/go/storage v1.25.0
+	cloud.google.com/go/storage v1.27.0
 	github.com/apache/airflow-client-go/airflow v0.0.0-20220509204651-4f1b26e4a5d0
 	github.com/aws/aws-sdk-go v1.40.33
+	github.com/databricks/databricks-sdk-go v0.1.1
```
Comment (nfx) [inline src/golang/go.mod:9]: v0.2.0 is a bit more consistent now

File: src/golang/lib/collections/integration/config.go [hunk 2]
```diff
@@ -45,6 +45,19 @@ type LambdaIntegrationConfig struct {
 	RoleArn string `json:"role_arn" yaml:"roleArn"`
 }
 
+type DatabricksIntegrationConfig struct {
+	// WorkspaceURL is the full url for the Databricks workspace that
+	// Aqueduct operators will run on.
+	WorkspaceURL string `json:"workspace_url" yaml:"workspaceUrl"`
+	// AccessToken is a Databricks AccessToken for a workspace. Information on how
+	// to create tokens can be found here: https://docs.databricks.com/dev-tools/auth.html#personal-access-tokens-for-users
+	AccessToken string `json:"access_token" yaml:"accessToken"`
```
Comment (nfx) [inline src/golang/lib/collections/integration/config.go:54]: please keep in mind, that Databricks has other authentication options. you could be copying those to this config, or just depend on `databricks.Config`. we don't have `yaml` field tags, though

File: src/golang/lib/databricks/constants.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package databricks
+
+const (
+	SparkVersion = "10.4.x-scala2.12"
```
Comment (nfx) [inline src/golang/lib/databricks/constants.go:4]: i'd suggest to use dynamic selectors: https://github.com/databricks/databricks-sdk-go#node-type-and-databricks-runtime-selectors

also DBR 10.x is a bit old now.

File: src/golang/lib/databricks/jobs.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	return createResp.JobId, nil
+}
+
+func RunNow(
+	ctx context.Context,
+	databricksClient *databricks_sdk.WorkspaceClient,
+	jobId int64,
+	specStr string,
+) (int64, error) {
+	runResp, err := databricksClient.Jobs.RunNow(
```
Comment (nfx) [inline src/golang/lib/databricks/jobs.go:90]: i'd suggest to use `RunNowAndWait`

https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.1.1/service/jobs#JobsAPI.RunNowAndWait

File: src/golang/lib/job/databricks.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+	getRunReq := &jobs.GetRun{
+		RunId: runId,
+	}
+	getRunResp, err := j.databricksClient.Jobs.GetRun(ctx, *getRunReq)
+	if err != nil {
+		return shared.UnknownExecutionStatus, systemError(errors.Wrap(err, "Unable to get run from databricks."))
+	}
+
+	switch getRunResp.State.LifeCycleState {
```
Comment (nfx) [inline src/golang/lib/job/databricks.go:137]: Would you want to decouple polling from RunNowAndWait on the Databricks SDK level?

https://pkg.go.dev/github.com/databricks/databricks-sdk-go@v0.1.1/service/jobs#JobsAPI.RunNowAndWait

**Comment by hsubbaraj-spiral** [comment]
Hi @nfx! Thanks for all the feedback you provided, I'll make the according revisions soon! A question from my end: Our project uses Go 1.16 (migrating to Go 1.18 is high on our roadmap), but we'd like to get this integration out using this SDK. Do you have any suggestions on how to go about doing this?

**Pull request merged**